src/web/rustdoc.rs

```diff
@@ -253,7 +253,11 @@ pub(crate) async fn rustdoc_redirector_handler(
         .into_response())
     } else {
         Ok(axum_cached_redirect(
-            format!("/crate/{crate_name}/{}", matched_release.req_version),
+            EscapedURI::new(
+                &format!("/crate/{crate_name}/{}", matched_release.req_version),
+                uri.query(),
```

Comment:
I'm missing a test for this updated behaviour here.

Reply:
I can't figure out what's the URL for it. I tried `/crate/sysinfo/0.29.0?search=q` and `/crate/sysinfo/0.40.0?search=q` and both have the query arguments even before this change. Any idea?

Reply:
the redirector is behind URLs like `/sysinfo`

Reply:
ah wait, you're searching for an example for this specific `else` part

Reply:
you will only reach it when `rustdoc_status` is `false`. 

That being said, should we update the `query_pairs` in the `redirect_to_doc` function in here too to your new `EscapedURI`?

Reply:
> you will only reach it when `rustdoc_status` is `false`.

Can't find a way to write a test with this case. Do we have a test with this situation by any chance?

Reply:
Nice, thanks a lot!